load_testing/justfile

```
@@ -1,12 +1,15 @@
-set dotenv-load := false
+set dotenv-load := true
 
 DOCKER_HOST := if os() == "macos" { "host.docker.internal" } else { "172.17.0.1" }
 
 build:
     docker build . -t openverse-load-testing:latest
 
 run script host: build
-    docker run --rm -it -v {{invocation_directory()}}:/app openverse-load-testing:latest ./{{ script }}.sh {{ host }}
+    docker run --rm -it -v {{ invocation_directory() }}:/app openverse-load-testing:latest ./{{ script }}.sh {{ host }}
 
 api host=("http://" + DOCKER_HOST + ":8000"):
     @just run api {{ host }}
+
+k6 env="-e API_TOKEN=$API_TOKEN": build
+    docker run --rm {{ env }} -v {{ invocation_directory() }}/k6:/app openverse-load-testing:latest k6 run main.js
```

This might be a slightly easier to use (more flexible) version:

```suggestion
k6 url="https://api-dev.openverse.engineering" +extra_args="": build
    docker run --rm -e API_TOKEN=$API_TOKEN -e API_URL={{ url }} -v {{ invocation_directory() }}/k6:/app {{ extra_args }} openverse-load-testing:latest k6 run main.js
```

What do you think?

---

I guess that the `env` can already serve to pass multiple arguments and not just environment variables 🤔 it just needs to be enclosed in "" like the code example in the first comment, but we could also pass arguments to k6 with:

```sh
k6 docker_args="-e API_TOKEN=$API_TOKEN" +k6_args="": build
    docker run --rm {{ docker_args }} -v {{ invocation_directory() }}/k6:/app openverse-load-testing:latest k6 run main.js {{ k6_args }}
```

Does it look better like this?

---

Well, if you decide to pass any extra args now, you'll have to also pass the API_TOKEN arg. Given that one _always_ needs to be passed, I was thinking we could remove it from the potential to be removed. If it needs to be overriden for some reason, I think `-e` arguments will override previous ones with the same name.

---

Ahh, I understand, then your suggestion is fine for me!